api/api_eth.go

```diff
@@ -1648,3 +1648,72 @@ func (api *EthAPI) Config(ctx context.Context) (*configResponse, error) {
 	}
 	return &resp, nil
 }
+
+func (api *EthAPI) GetBlobSidecars(ctx context.Context, number *rpc.BlockNumber, fullBlob bool) ([]*map[string]interface{}, error) {
+	if *number == rpc.PendingBlockNumber {
+		return nil, errors.New("pending block not supported")
+	}
+	block, err := api.kaiaBlockChainAPI.b.BlockByNumber(ctx, *number)
+	if err != nil {
+		return nil, err
+	}
+
+	results := make([]*map[string]interface{}, eip4844.MaxBlobsPerBlock(api.kaiaBlockChainAPI.b.ChainConfig(), block.Number()))
+	count := 0
+	for txIndex, tx := range block.Transactions() {
+		if tx.Type() != types.TxTypeEthereumBlob {
+			continue
+		}
+		sidecar, err := api.kaiaBlockChainAPI.b.GetBlobSidecar(block.Number(), txIndex, tx.Hash())
+		if err != nil {
+			return nil, err
+		}
+
+		results[count] = api.RpcMarshalBlobSidecar(sidecar, fullBlob, block.Hash(), block.Number(), tx.Hash(), txIndex)
+		count++
+	}
+
+	return results[:count], nil
```

Comment:
MaxBlobsPerBlock should be capacity, not length?

```suggestion
	results := make([]*map[string]interface{}, 0, eip4844.MaxBlobsPerBlock(api.kaiaBlockChainAPI.b.ChainConfig(), block.Number()))
	for txIndex, tx := range block.Transactions() {
		if tx.Type() != types.TxTypeEthereumBlob {
			continue
		}
		sidecar, err := api.kaiaBlockChainAPI.b.GetBlobSidecar(block.Number(), txIndex, tx.Hash())
		if err != nil {
			return nil, err
		}

		results = append(results, api.RpcMarshalBlobSidecar(sidecar, fullBlob, block.Hash(), block.Number(), tx.Hash(), txIndex))
	}

	return results, nil
```